src/ServiceProvider.php

```diff
@@ -29,33 +27,43 @@ public function boot(): void
 
     public function register(): void
     {
-        $client = fn () => new Client([
+        $this->registerClasses();
+    }
+
+    protected function createGuzzleClient(): Client
+    {
+        return new Client([
             'base_uri' => config('features.flagr_url')
         ]);
+    }
 
-        $this->app->bind(Feature::class, function () use ($client) {
+    protected function registerClasses(): void
```

Comment:
**nitpick**
```suggestion
    private function registerClasses(): void
```

Reply:
That was a deliberate choice to allow any future users to override